crates/core/component/dex/src/component/position_manager.rs

```diff
@@ -428,6 +432,53 @@ pub(crate) trait Inner: StateWrite {
         Ok(())
     }
 
+    // TODO(erwan): break this out into a `position_manager::inventory_index` module.
+    fn index_position_by_inventory(&mut self, position: &position::Position, id: &position::Id) {
+        tracing::debug!("indexing position by inventory");
+        let canonical_pair = position.phi.pair;
+        // A position is bound to an unordered trading pair: A <> B.
+        // We want to index the position by inventory for each direction:
+        // A -> B
+        let pair_ab = DirectedTradingPair::new(canonical_pair.asset_1(), canonical_pair.asset_2());
+        let inventory_a = position
+            .reserves_for(pair_ab.start)
+            .expect("infaillible (remove it?)");
```

Comment:
should we address these expect statements here? would the infallible `reserves_1` and `reserves_2` methods be useful for this?

Reply:
Good catch, using `reserves_for` here is more defensive because we are dealing with two levels of semantics wrt ordering:
- a `TradingPair` has a canonical ordering based on the random field elements that make up its two asset `Id`s
- a `DirectedTradingPair` is an ordered trading pair i.e. one that has both a canonical ordering and an application-level one


The `reserves_1` and `reserves_2` getters return the reserves relative to the canonical ordering, but it might not correspond to the one we want. We could write the method so that they do always coincide, I have considered it, but this breaks silently during a refactor - so more brittle than an extra branch.

Reply:
ah! thanks for the explanation 🙂sounds good to me